### PR TITLE
Guide. Step four

### PR DESCRIPTION
**Pull request by dmdashenkov**
The fourth step of the TodoList guide.

See #55 for more details.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: api-java/src/main/java/io/spine/examples/todolist/repository/TaskLabelsRepository.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.todolist.repository;
```
Comment (armiol) [inline api-java/src/main/java/io/spine/examples/todolist/repository/TaskLabelsRepository.java]: Let's discuss the packaging. I think it's fine for a tutorial, but having "events", "repository" etc is not what we recommend. We prefer grouping the classes by their meaning. E.g. "task", "label" etc.

I think we need to create an issue to repackage the whole thing in future.

File: model/src/main/proto/todolist/values.proto [hunk 2]
```diff
@@ -154,3 +197,17 @@ message RejectedTaskCommandDetails {
...
+
+// DTO for the unsuccessful label command details.
+message RejectedLabelCommandDetails {
+
+    // A task label identifier.
+    LabelId label_id = 1;
+}
+
+// DTO for the list of the label IDs.
+message LabelIdsList {
```
Comment (armiol) [inline model/src/main/proto/todolist/values.proto]: It should be `LabelIdList`. No "s".

File: testutil-api/src/main/java/io/spine/examples/todolist/testdata/TestTaskLabelsCommandFactory.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import io.spine.examples.todolist.TaskId;
+import io.spine.examples.todolist.c.commands.AssignLabelToTask;
+import io.spine.examples.todolist.c.commands.RemoveLabelFromTask;
+
+/**
+ * A factory of the task label commands for the test needs.
+ *
+ * @author Illia Shepilov
+ */
+public class TestTaskLabelsCommandFactory {
```
Comment (armiol) [inline testutil-api/src/main/java/io/spine/examples/todolist/testdata/TestTaskLabelsCommandFactory.java]: We should migrate the whole thing to `TestEnv`.

This is another issue.

File: api-java/src/main/java/io/spine/examples/todolist/repository/TaskLabelsRepository.java (hunk 1 above)
Comment (dmdashenkov) [reply api-java/src/main/java/io/spine/examples/todolist/repository/TaskLabelsRepository.java]: Issue #59.

File: testutil-api/src/main/java/io/spine/examples/todolist/testdata/TestTaskLabelsCommandFactory.java (hunk 3 above)
Comment (dmdashenkov) [reply testutil-api/src/main/java/io/spine/examples/todolist/testdata/TestTaskLabelsCommandFactory.java]: Issue #60.

**Pull request merged**